File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Editor/Multilingual/tools/protocol/t_WindowBin.py
from CVariable import CReadVariable,CWriteVariable
from Vector import *
from t_WindowData import	t_WindowData
#/==========================================================================
#*!
#	@brief	t_WindowBin
#/
class t_WindowBin:
	#/==========================================================================
	#*!
	#	@brief	Member
	#/
	m_name = ""	#std::string
	m_exportpath = ""	#std::string
	m_mAssetbundle = 0	#U32
	m_data = t_WindowData()	#t_WindowData
	#/==========================================================================
	#*!
	#	@brief	Constructor
	#/
	def __init__(self):
		self.clear()

	#/==========================================================================
	#*!
	#	@brief	Accessor
	#/
	def clear(self):
		self.m_name = ""
		self.m_exportpath = ""
		self.m_mAssetbundle = 0
		self.m_data = t_WindowData()

	def read(self,cVariable):
		self.m_name = cVariable.getString(255)
		self.m_exportpath = cVariable.getString(255)
		self.m_mAssetbundle = cVariable.getU32()
		if (not self.m_data.read(cVariable)):
			return False
		return True

	def write(self,cVariable):
		cVariable.putString(self.m_name,255)
		cVariable.putString(self.m_exportpath,255)
		cVariable.putU32(self.m_mAssetbundle)
		if (not self.m_data.write(cVariable)):
			return False
		return True

File: Make Human Unity/Assets/MakeHuman/Icons/KsSoft/Plugins/Protocol/t_WindowBin.cs
//==============================================================================================
/*!
	@file  KsSoft Protocol
	
	(counter SJIS string 京)
*/
//==============================================================================================
using UnityEngine;
using System.Collections;
using System.Collections.Generic;

using KS;
//==========================================================================
/*!
	@brief	t_WindowBin
*/
public class t_WindowBin : ISerializable {
	//==========================================================================
	/*!
		@brief	Member
	*/
	public string	m_name;
	public string	m_exportpath;
	public uint	m_mAssetbundle;
	public t_WindowData	m_data;
	//==========================================================================
	/*!
		@brief	Constructor
	*/
	public t_WindowBin() {
		m_data = new t_WindowData();
		clear();
	}
	//==========================================================================
	/*!
		@brief	Accessor
	*/
	public void clear() {
		m_name = "";
		m_exportpath = "";
		m_mAssetbundle = 0;
		m_data.clear();
	}
	public bool read(CReadVariable cVariable) {
		try {
			cVariable.getString(ref m_name,255);
			cVariable.getString(ref m_exportpath,255);
			m_mAssetbundle = cVariable.getU32();
			if (!m_data.read(cVariable)) return false;
		} catch (System.Exception e) {
			Debug.LogError(e);return false;
		}
		return true;
	}
	public bool write(CWriteVariable cVariable) {
		cVariable.put(ref m_name,255);
		cVariable.put(ref m_exportpath,255);
		cVariable.put(m_mAssetbundle);
		if (!m_data.write(cVariable)) return false;
		return true;
	}
};

